src/GoodsList.tsx

```diff
@@ -2,15 +2,15 @@ import React from 'react';
 import { Good } from './types/Good';
 
 type Props = {
-  goods: Good[]
+  goods: Good[];
 };
 
-export const GoodsList: React.FC<Props> = ({ goods }) => (
+export const GoodsList: React.FC<Props> = React.memo(({ goods }) => (
```

Comment:
React memo redindant here, because goods changed every time you click on the button(getAll, get5...)